### PR TITLE
ref(hooks): Remove params + reactCreateClass from HookOrDefault

### PR DESCRIPTION
**Pull request by evanpurkhiser**
pretty much just cleanup

### CODE REVIEW

File: static/app/components/hookOrDefault.tsx [hunk 1]
```diff
@@ -56,28 +46,28 @@ function HookOrDefault<H extends HookName>({
...
+      hooks: HookStore.get(hookName),
+    };
 
-    handleHooks(hookNameFromStore: HookName, hooks: Array<Hooks[H]>) {
-      // Make sure that the incoming hook update matches this component's hook name
-      if (hookName !== hookNameFromStore) {
-        return;
-      }
+    componentWillUnmount() {
+      this.unlistener?.();
```
Comment (k-fish) [inline static/app/components/hookOrDefault.tsx:61]: I like calling it `unlistener`, nice.

File: static/app/components/hookOrDefault.tsx [hunk 2]
```diff
@@ -20,14 +18,6 @@ type Params<H extends HookName> = {
    * use React.Suspense and React.lazy to render the component.
    */
   defaultComponentPromise?: () => Promise<ReturnType<Hooks[H]>>;
-  /**
-   * Parameters to pass into the hook callback
-   */
-  params?: Parameters<Hooks[H]>;
```
Comment (EvanPurkhiser) [inline static/app/components/hookOrDefault.tsx:26]: We never use this, and honestly we should just change these hooks so they're not functions.

File: static/app/components/hookOrDefault.tsx [hunk 3]
```diff
@@ -90,18 +80,20 @@ function HookOrDefault<H extends HookName>({
...
       const hookExists = this.state.hooks && this.state.hooks.length;
+      const componentFromHook = this.state.hooks[0]();
+
       const HookComponent =
-        hookExists && this.state.hooks[0]({params})
-          ? this.state.hooks[0]({params})
-          : this.getDefaultComponent();
+        hookExists && componentFromHook ? componentFromHook : this.defaultComponent;
+
+      return HookComponent ? <HookComponent {...this.props} /> : null;
```
Comment (EvanPurkhiser) [inline static/app/components/hookOrDefault.tsx:92]: This fixes the typing so that you _can_ have a empty defaultComponent.

Comment (k-fish) [inline static/app/components/hookOrDefault.tsx:92]: Can't we change `defaultComponent` to optional then?

**Review by k-fish — APPROVED** [review]

File: static/app/components/hookOrDefault.tsx (hunk 3 above)
Comment (EvanPurkhiser) [reply static/app/components/hookOrDefault.tsx:92]: it already is

Comment (k-fish) [reply static/app/components/hookOrDefault.tsx:92]: yeah just saw that, lol.

**Pull request merged**